src/DisplayContainer.tsx

```diff
@@ -1,14 +1,18 @@
-import { memo } from "react";
+import { memo, useMemo } from "react";
 import Display from "./Display";
 import { Name } from "./App";
 
 const DisplayContainer = ({name}: {name: Name}) => {
 
+  const firstName = useMemo(() => `${name.first}`, [name.first]);
+  const lastName = useMemo(() => `${name.last}`, [name.last]);
```

Comment:
The slowness is from the comment I left above since we want first and last name to update on change. The useMemo wound't do anything here since the name prop is updating on each keystroke. When props change the component re-renders no matter what.